closed/src/java.base/share/classes/com/sun/crypto/provider/NativeChaCha20Cipher.java

```diff
@@ -87,20 +89,47 @@
     // The underlying engine for doing the ChaCha20/Poly1305 work
     private ChaChaEngine engine;
 
-    private static NativeCrypto nativeCrypto;
+    private static final NativeCrypto nativeCrypto;
+    private static final Cleaner contextCleaner;
     private long context;
 
     private final ByteArrayOutputStream aadBuf;
 
     static {
         nativeCrypto = NativeCrypto.getNativeCrypto();
+        contextCleaner = CleanerFactory.cleaner();
+    }
+
+    private static final class ChaCha20CleanerRunnable implements Runnable {
+        private final long context;
+
+        public ChaCha20CleanerRunnable(long context) {
+            this.context = context;
+        }
+
+        @Override
+        public void run() {
+            /*
+             * Release the ChaCha20 context.
+             */
+            long ret = nativeCrypto.DestroyContext(context);
+            if (ret == -1) {
+                throw new ProviderException("Error in NativeChaCha20Cipher - DestroyContext");
+            }
```

Comment:
There's no point in throwing an exception that will just be ignored. This should just be
```
    nativeCrypto.DestroyContext(context);
```

Reply:
updated